src/driver/kvm.rs

```diff
@@ -85,12 +86,94 @@ impl Introspectable for Kvm {
             rip: regs.rip,
             rflags: regs.rflags,
             cr0: sregs.cr0,
+            cr2: sregs.cr2,
             cr3: sregs.cr3,
             cr4: sregs.cr4,
-            fs_base: sregs.fs.base,
+            sysenter_cs: _msrs.entries[0].data,
+            sysenter_esp: _msrs.entries[1].data,
+            sysenter_eip: _msrs.entries[2].data,
+            msr_efer: _msrs.entries[3].data,
+            msr_star: _msrs.entries[4].data,
+            msr_lstar: _msrs.entries[5].data,
+            efer: sregs.efer,
+            apic_base: sregs.apic_base,
+            cs: segment_reg {
+                base: sregs.cs.base,
+                limit: sregs.cs.limit,
+                selector: sregs.cs.selector,
+            },
+            ds: segment_reg {
+                base: sregs.ds.base,
+                limit: sregs.ds.limit,
+                selector: sregs.ds.selector,
+            },
+            es: segment_reg {
+                base: sregs.es.base,
+                limit: sregs.es.limit,
+                selector: sregs.es.selector,
+            },
+            fs: segment_reg {
+                base: sregs.fs.base,
+                limit: sregs.fs.limit,
+                selector: sregs.fs.selector,
+            },
+            gs: segment_reg {
+                base: sregs.gs.base,
+                limit: sregs.gs.limit,
+                selector: sregs.gs.selector,
+            },
+            ss: segment_reg {
+                base: sregs.ss.base,
+                limit: sregs.ss.limit,
+                selector: sregs.ss.selector,
+            },
+            tr: segment_reg {
+                base: sregs.tr.base,
+                limit: sregs.tr.limit,
+                selector: sregs.tr.selector,
+            },
+            ldt: segment_reg {
+                base: sregs.ldt.base,
+                limit: sregs.ldt.limit,
+                selector: sregs.ldt.selector,
+            },
         }))
     }
 
+    fn write_registers(&self, vcpu: u16, reg: Registers) -> Result<(), Box<dyn Error>> {
+        let register: X86Registers;
+        match reg {
+            Registers::X86(x86_registers) => {
+                register = x86_registers;
+            }
+        }
+
+        let regs = KVMiRegs {
```

Comment:
nitpick: You could also use the From trait here which would reduce the amount of lines. Then you could put everything into the match arm above.